libbeat/autodiscover/autodiscover.go

```diff
@@ -176,18 +198,15 @@ func (a *Autodiscover) handleStop(event bus.Event) {
 			continue
 		}
 
-		if runner := a.runners.Get(hash); runner != nil {
-			// Stop can block, we run it asyncrhonously to avoid blocking
-			// the whole events loop. The runner hash is removed in any case
-			// so an equivalent configuration can be added again while this
-			// one stops, and a duplicated event don't try to stop it twice.
-			logp.Info("Autodiscover stopping runner: %s", runner)
-			go runner.Stop()
-			a.runners.Remove(hash)
+		if a.runners.Has(hash) {
+			delete(a.configs, hash)
```

Comment:
Can only 1 go routine at the time access `a.configs`?

Where is the runner stopped now?

Reply:
configs is only modified by the event loop (`handleStart` and `handleStop`), so there is only one goroutine using it. Start/Stop is now handled by `runners.Reload`: https://github.com/elastic/beats/pull/7172/files/63c350a7ea841fd6f2b5bea7bea4debe1c0afdf5#diff-3b0ae03a54b2e31c58f8dcf308a6a7b8R127. This is the same code path used by configuration reload mechanism